app/models/exam.rb

```diff
@@ -1,19 +1,28 @@
 class Exam < ApplicationRecord
+
   include GuideContainer
   include FriendlyName
-
-  validates_presence_of :start_time, :end_time
+  include TerminalNavigation
 
   belongs_to :organization
+  belongs_to :course, optional: true
 
   has_many :authorizations, class_name: 'ExamAuthorization', dependent: :destroy
   has_many :users, through: :authorizations
 
+  enum passing_criterion_type: [:none, :percentage, :passed_exercises], _prefix: :passing_criterion
+
+  validates_presence_of :start_time, :end_time
+  validates_numericality_of :max_problem_submissions, :max_choice_submissions, greater_than_or_equal_to: 1, allow_nil: true
+
+  before_save :set_default_criterion_type!
+  before_save :ensure_valid_passing_criterion!
+
+  before_create :set_classroom_id!
```

Comment:
:thinking: 

I am not quite sure about the semantics of a _classroom id_ in domain. Why not calling it `code`?

Reply:
Mmm I agree with you but classroom_id it's already a field in domain :stuck_out_tongue: 
We should change it in the next PR, that's true